src/main/java/org/bytedeco/javacv/FFmpegFrameGrabber.java

```diff
@@ -1303,19 +1313,8 @@ public synchronized Frame grabFrame(boolean doAudio, boolean doVideo, boolean do
 
         boolean videoFrameGrabbed = frameGrabbed && frame.image != null;
         boolean audioFrameGrabbed = frameGrabbed && frame.samples != null;
+        boolean dataFrameGrabbed = frameGrabbed && frame.data != null;
         frameGrabbed = false;
-        frame.keyFrame = false;
-        frame.imageWidth = 0;
-        frame.imageHeight = 0;
-        frame.imageDepth = 0;
-        frame.imageChannels = 0;
-        frame.imageStride = 0;
-        frame.image = null;
-        frame.sampleRate = 0;
-        frame.audioChannels = 0;
-        frame.samples = null;
-        frame.data = null;
-        frame.opaque = null;
```

Comment:
What's wrong with resetting the Frame here?

Reply:
If the reset appears earlier than checking videoFrameGrabbed (audioFrameGrabbed, dataFrameGrabbed), then we return a frame with default values if they are true.

Reply:
Oh, I see, there was a bug there.